signals/src/main/java/com/vaadin/signals/impl/AsyncSignalTree.java

```diff
@@ -0,0 +1,235 @@
+package com.vaadin.signals.impl;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import com.vaadin.signals.Id;
+import com.vaadin.signals.SignalCommand;
+import com.vaadin.signals.SignalCommand.SnapshotCommand;
+
+/**
+ * A signal tree that submits commands to an event log and asynchronously waits
+ * for external confirmation before completing handling of the command. This
+ * means that {@link #confirmed} may contain changes that are not yet in
+ * {@link #submitted} and might never end up there if a concurrent change causes
+ * a conflict. This type of tree is intended for signals that are synchronized
+ * across a cluster.
```

Comment:
As the submitted commands may not be confirmed yet, the Javadoc should say:
```suggestion
 * A signal tree that submits commands to an event log and asynchronously waits
 * for external confirmation before completing handling of the command. This
 * means that {@link #submitted} may contain changes that are not yet in
 * {@link #confirmed} and might never end up there if a concurrent change causes
 * a conflict. This type of tree is intended for signals that are synchronized
 * across a cluster.
```